### ISSUE TITLE
Different predictions using DNN module.

### ISSUE THREAD

**Issue by chrisrn** [opencv/opencv#9108]
OS: Ubuntu 16.04 LTS
g++ version: 5.4.0

I have already trained cifar-net model using tensorflow. The preprocessing of the training images has been done by opencv. The only tensorflow preprocessing is the tf.image.per_image_standardization function which normalizes the image by subtracting by the mean and dividing by the variance. So during prediction with tf_inception script from your repo, I did this preprocessing using opencv and the pixel values are the same, so the input of the network is the same as the training process. The problem is that the prediction is wrong and I cannot understand why. Steps I followed:

1) Training and exporting .pbtxt file
2) Freeze graph using the training checkpoint.
3) Feed images and graph to tf_inception script.

The accuracy using python and TF on a batch of images is 95% but with c++ and opencv is 80%. The basic operations that cifar-net uses are "Conv2D", "MaxPool" and "LRN". By feeding the same images to both TF and opencv predictors, I wonder if there are any differences between your layers' implementation and TF's implementation.

**Comment by dkurt** [comment]
@chrisrn, please, could you upload your model somewhere to let us reproduce the problem. In example by Google Drive. Thanks!

**Comment by chrisrn** [comment]
You mean the protocol buffer file?

**Comment by dkurt** [comment]
Yeah, if it possible. Because I'm going to try to run this model both using TensorFlow and DNN and compare outputs.

**Comment by chrisrn** [comment]
Here it is:
https://drive.google.com/open?id=0BwgbAIs43uIoYjNsdlRDa0Etb3c

**Comment by dkurt** [comment]
@chrisrn , I've tried to load model in TF and got output for the random input of batch size 1. Then I've compared it with an output of DNN for the same input. Results:
```
TF
9.809610247611999512e-01
1.903890632092952728e-02

DNN
0.980961
0.0190389
```
So model imports fine in my case.
Please, could you try to localize your problem? Get an image / batch of images which predictions are different for DNN and TF?

**Comment by chrisrn** [comment]
I will try to find some examples. But if the model is right, then probably the problem has to do with preprocessing. Because during training I am using:
```
tf.image.per_image_standardization(image)
```

and in DNN I use:
```
Mat mean, stddev;
double imageMean, imageStddev;
meanStdDev(image, mean, stddev);
imageMean = mean.at<double>(0);
imageStddev = stddev.at<double>(0);
subtract(image, imageMean, image);
image = image / imageStddev;
```
The strange part is that I have already checked the pixel values using both TF and DNN standardization and they are the same.

**Comment by chrisrn** [comment]
Problem solved, it was something irrelevant. Dnn module is ok. I close the topic.

**Comment by Dxye** [comment]
@chrisrn  Do you remember the reason why dnn predictions different from the tf predictions using same model?

**Comment by chrisrn** [comment]
The problem had to do with the ground truths and not with the predictions. I forgot to correct an if statement to acquire them correctly. So the dnn module has no problem and the work-flow is the one that I recommended.
